scripts/curiosity.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+PKG = 'embodied_attention'
+import roslib; roslib.load_manifest(PKG)
+
+import sys
+import rospy
+from std_msgs.msg import Float64, Float32MultiArray, MultiArrayLayout, MultiArrayDimension
+from sensor_msgs.msg import Image, CameraInfo, JointState
+from image_geometry import StereoCameraModel
+from stereo_msgs.msg import DisparityImage
+from geometry_msgs.msg import Point
+from tf2_geometry_msgs import PointStamped
+from cv_bridge import CvBridge, CvBridgeError
+import numpy as np
+import tf2_ros
+import traceback
+from skimage.draw import circle
+
+class Curiosity():
+    def __init__(self):
+        saliency_map_sub = rospy.Subscriber("/saliency_map", Float32MultiArray, self.saliency_map_callback, queue_size=1, buff_size=2**24)
+        camera_info_left_sub = rospy.Subscriber("/hollie/camera/left/camera_info", CameraInfo, self.camera_info_left_callback, queue_size=1, buff_size=2**24)
+        camera_info_right_sub = rospy.Subscriber("/hollie/camera/right/camera_info", CameraInfo, self.camera_info_right_callback, queue_size=1, buff_size=2**24)
+        point_sub = rospy.Subscriber("/saccade_point", PointStamped, self.saccade_point_callback, queue_size=1, buff_size=2**24)
+        disparity_sub = rospy.Subscriber("/hollie/camera/disparity", DisparityImage, self.disparity_callback, queue_size=1, buff_size=2**24)
+
+        self.saliency_map_curiosity_pub = rospy.Publisher("/saliency_map_curiosity", Float32MultiArray, queue_size=1)
+        self.saliency_map_curiosity_image_pub = rospy.Publisher("/saliency_map_curiosity_image", Image, queue_size=1)
+
+        self.camera_info_left = None
+        self.camera_info_right = None
+        self.disparity_image = None
+        self.camera_model = StereoCameraModel()
+
+        self.targets = []
+
+        self.cv_bridge = CvBridge()
+
+        self.saliency_width = float(rospy.get_param('~saliency_width', '256'))
+        self.saliency_height = float(rospy.get_param('~saliency_height', '192'))
+        self.min_disparity = rospy.get_param("/hollie/camera/stereo_image_proc/min_disparity", "-16")
+
+        self.tfBuffer = tf2_ros.Buffer(rospy.Duration(30))
+        self.listener = tf2_ros.TransformListener(self.tfBuffer)
+
+    def saliency_map_callback(self, saliency_map):
+        if self.camera_info_left is not None and self.camera_info_right is not None and self.disparity_image is not None:
+            # handle input
+            lo = saliency_map.layout
+            saliency_map = np.asarray(saliency_map.data[lo.data_offset:]).reshape(lo.dim[0].size, lo.dim[1].size)
+
+            # modify saliency map
+            self.camera_model.fromCameraInfo(self.camera_info_left, self.camera_info_right)
+            disparity_image = self.cv_bridge.imgmsg_to_cv2(self.disparity_image.image)
```

Comment:
the curiosity needs the disparity? I'm not sure I understand why, is it to get the depth of objects & the activity shift?